server/src-lib/Hasura/RQL/Types/CustomTypes.hs

```diff
@@ -141,39 +138,49 @@ data TypeRelationship t f
 instance (NFData t, NFData f) => NFData (TypeRelationship t f)
 instance (Cacheable t, Cacheable f) => Cacheable (TypeRelationship t f)
 $(makeLenses ''TypeRelationship)
-
-type TypeRelationshipDefinition =
-  TypeRelationship QualifiedTable PGCol
-
 $(J.deriveJSON (J.aesonDrop 3 J.snakeCase) ''TypeRelationship)
 
 newtype ObjectTypeName
-  = ObjectTypeName { unObjectTypeName :: G.NamedType }
+  = ObjectTypeName { unObjectTypeName :: G.Name }
   deriving ( Show, Eq, Ord, Hashable, J.FromJSON, J.FromJSONKey, DQuote
            , J.ToJSONKey, J.ToJSON, Lift, Generic, NFData, Cacheable)
 
-data ObjectTypeDefinition
+data ObjectTypeDefinition a b c
   = ObjectTypeDefinition
   { _otdName          :: !ObjectTypeName
   , _otdDescription   :: !(Maybe G.Description)
-  , _otdFields        :: !(NEList.NonEmpty ObjectFieldDefinition)
-  , _otdRelationships :: !(Maybe [TypeRelationshipDefinition])
+  , _otdFields        :: !(NonEmpty (ObjectFieldDefinition a))
+  , _otdRelationships :: !(Maybe (NonEmpty (TypeRelationship b c)))
   } deriving (Show, Eq, Lift, Generic)
-instance NFData  ObjectTypeDefinition
-instance Cacheable   ObjectTypeDefinition
+instance (NFData a, NFData b, NFData c) => NFData (ObjectTypeDefinition a b c)
+instance (Cacheable a, Cacheable b, Cacheable c) => Cacheable (ObjectTypeDefinition a b c)
 $(J.deriveJSON (J.aesonDrop 4 J.snakeCase) ''ObjectTypeDefinition)
 
 data ScalarTypeDefinition
   = ScalarTypeDefinition
-  { _stdName        :: !G.NamedType
+  { _stdName        :: !G.Name
   , _stdDescription :: !(Maybe G.Description)
   } deriving (Show, Eq, Lift, Generic)
 instance NFData ScalarTypeDefinition
 instance Cacheable ScalarTypeDefinition
+instance Hashable ScalarTypeDefinition
 $(J.deriveJSON (J.aesonDrop 4 J.snakeCase) ''ScalarTypeDefinition)
 
+-- default scalar names
+intScalar, floatScalar, stringScalar, boolScalar, idScalar :: G.Name
+intScalar    = $$(G.litName "Int")
+floatScalar  = $$(G.litName "Float")
+stringScalar = $$(G.litName "String")
+boolScalar   = $$(G.litName "Boolean")
+idScalar     = $$(G.litName "ID")
+
+defaultScalars :: [ScalarTypeDefinition]
+defaultScalars =
+  map (flip ScalarTypeDefinition Nothing)
```

Comment:
I think it'll be a good idea to have a description here instead of `Nothing` for description.

Reply:
Perhaps we can similarly fill in the descriptions for the scalars defined in `Hasura.GraphQL.Parser.Internal.Parser`, and reuse that type information here?

Reply:
There are no descriptions for scalars in `Hasura.GraphQL.Parser.Internal.Parser` at the moment; we'd need to decide how they should be worded, if at all. I'll use the scalar name definitions in the parsers, though.